### PR TITLE
[tools] Show helpful message when demangling '_'

### PR DESCRIPTION
**Pull request by kastiglione**
From most shells, the symbol(s) given to `swift demangle` should be quoted or escaped, otherwise the shell will perform variable expansion on the `$...` suffix of the symbol. For example:

```
% swift demangle _$sScMMa
_ ---> _
```

Instead the command should be edited to one of the following:
```
% swift demangle sScMMa
_$sScMMa ---> type metadata accessor for Swift.MainActor
% swift demangle '_$sScMMa'
_$sScMMa ---> type metadata accessor for Swift.MainActor
% swift demangle _\$sScMMa
_$sScMMa ---> type metadata accessor for Swift.MainActor
```

When variable expansion has happened, `swift demangle` will see the symbol only as "`_`". This change adds a warning that identifies the likely issue, and contains suggestions on how the user can fix and rerun the invocation.

### CODE REVIEW

File: tools/swift-demangle/swift-demangle.cpp [hunk 1]
```diff
@@ -391,6 +391,13 @@ int main(int argc, char **argv) {
   } else {
     swift::Demangle::Context DCtx;
     for (llvm::StringRef name : InputNames) {
+      if (name == "_") {
+        llvm::errs()
+            << "warning: input symbol '_' is likely the result of variable "
+               "expansion by the shell. Either remove the '_$' prefix, single "
+               "quote the symbol, or escape the '$' with a backslash";
```
Comment (kastiglione) [inline tools/swift-demangle/swift-demangle.cpp:398]: Any thoughts on the suggestions? Some of these might be shell specific.

**Review by mikeash — APPROVED** [review]
Great idea!

File: tools/swift-demangle/swift-demangle.cpp (hunk 1 above)
Comment (kastiglione) [reply tools/swift-demangle/swift-demangle.cpp]: update

**Comment by kastiglione** [comment]
@swift-ci smoke test

**Review by al45tair — APPROVED** [review]
Neat idea — like it :-)

**Pull request merged**